ipapython/admintool.py

```diff
@@ -136,6 +136,8 @@ def main(cls, argv):
         :param argv: Command-line arguments.
         :return: Command exit code
         """
+        standard_logging_setup(None, verbose=True)
```

Comment:
This is called later in the execution.
Your so-called proper fix is no proper fix at all, you should have realized that something's wrong when @frasertweedale showed you the doubled output.
Please fix this properly and don't call `standard_logging_setup()` twice.

Reply:
I've been a bit prickly in my previous comment, sorry. Please, find another way to do this.

Reply:
Hi @stlaz, thanks for reviewing this. Do you think that is so wrong call `standard_logging_setup` another time? IMHO it's harmless. Actually, it is already called twice*, my patch would call it a third time.

Maybe, a better fix would be call `self._setup_logging(no_file=True)`, however `main` is a class method, so it's not possible call `_setup_logging`.

* It's called twice because in `execute` method, it calls `self._setup_logging(no_file=True)` and some lines bellow `self.setup_logging()`. Both methods calls `standard_logging_setup`.

Reply:
It is wrong and what you’re pointing out is a hack. I’ll post more tomorrow when I’m not in a cinema :)

Reply:
Here I go as promised.

As you can see, previously, we did not allow logging during the handling of the options. That's there for a good reason - some options are able to influence how and where to log. Therefore the only issue you're seeing in your PR https://github.com/freeipa/freeipa/pull/903 is nothing but the one you introduced yourself.

This is rather magical and is not documented anywhere. That's wrong. I did not realize that sooner either.

I would propose fixing https://github.com/freeipa/freeipa/pull/903 to print warning in runtime since we're not able to "fix" this properly (I don't think it's actually broken, it may only be way too magical).

Reply:
Hmm ok, now I understand your reasons. Ok, so I updated 903 and I'll close this as rejected, ok?

Reply:
Just bringing more info to the table (to future discussions) as was discussed on IRC:

The getLogger method will always return only one instance of the log, it will not create another object, as we can see in the docs [1]: "*All calls to this function with a given name return the same logger instance. This means that logger instances never need to be passed between different parts of an application.*" 

So, every call to `standard_logging_setup` would reinitialize the config was already done, due to `root_logger.handlers = []`. This can lead to an overload, but it seems harmless IMHO. 

In the future, we can try to find the options are able to influence how and where to log (as standa said) and make them independent.

[1] https://docs.python.org/2/library/logging.html#logging.getLogger